### DISCUSSION TITLE
2주차 Discussion

### DISCUSSION THREAD

**Discussion by LenKIM** [slippStudy/java-network-study#5] [General]
## 📢 책을 읽으면서 궁금한 점들을 자유롭게 적고 대답해 주세요

2주차 진행 방식

> 진행자: 진욱님
> 

- 3장 ~ 4장 120p 까지 읽기
- Discussion 에 질문 적어오기

**Comment by LenKIM** [comment]
어떤 자바 IO에 대한 API 는 어떤 것들이 있는가?

**Comment by LenKIM** [comment]
자바 IO 를 활용한 방식과 비슷한 API 가 무엇이 있을까?

자바의 상속을 사용하는 예시는 무엇이 있는지?

**Reply by LenKIM** [reply]
커스텀한 IO 을 만들 수 있는건가 그럼?

**Comment by jinwookh** [comment]
1. 3장을 보면 부모와 child가 동일한 변수를 사용하는 케이스가 나온다.   
실제로 이렇게 사용하는 케이스가 있을까?

2. 이번에 IO를 공부하면서 느낀 점은, NIO는 왠지 IO의 대체재가 아닌 것 같다는 점.. NIO는 IO를 보완하는 느낌이지 않을까 싶다.
실제로도 그럴까?

**Reply by chloe-codes1** [reply]
1번 질문 관련 참고

[Child class fields should not shadow parent class fields](https://rules.sonarsource.com/java/RSPEC-2387/)

**Comment by yangyunhyeuk** [comment]
1. System.out.println() 사용을 지양해야 하는 이유는 무엇인가?
2. InputStream, OutputStream을 사용하며, 문제가 발생했던 상황이 있었는가?

**Comment by chloe-codes1** [comment]
1. 3장에서 _"Java IO는 객체지향의 특성이 잘 내포되어 있다"_ 라고 한다. 그렇다면 Java IO는 어떻게 OOP 원칙을 따르고 있을까?
2. ByteStream, CharacterStream을 각각 어떤 상황에서 쓰면 효율적일까?

**Comment by kangbada0728** [comment]
1. PipedStream 이 네트워크에서 사용된다고 하는데 정확히 어떻게 사용될까.
2. BufferedInputStream 과 read(buffer) 의 차이점은 무엇일까.
